pkg/controller/database/cloudsql.go

```diff
@@ -245,3 +247,24 @@ func getConnectionDetails(cr *v1beta1.CloudSQLInstance, instance *sqladmin.Datab
 
 	return m
 }
+
+type cloudsqlTagger struct {
+	kube client.Client
+}
+
+// Initialize adds the external tags to spec.forProvider.settings.userLabels
+func (t *cloudsqlTagger) Initialize(ctx context.Context, mg resource.Managed) error {
+	cr, ok := mg.(*v1beta1.CloudSQLInstance)
+	if !ok {
+		return errors.New(errNotCloudSQL)
+	}
+	if cr.Spec.ForProvider.Settings.UserLabels == nil {
+		cr.Spec.ForProvider.Settings.UserLabels = map[string]string{}
+	}
+	tags := resource.GetExternalTags(cr)
+	for k, v := range tags {
```

Comment:
Nit: this could be `for k, v := range resource.GetExternalTags(cr)`

Reply:
I was thinking that the function would be called repeatedly, seems like that's not the case. Thanks!